### DISCUSSION TITLE
Thanos Query Sidecar Endpoint specific metric?

### DISCUSSION THREAD

**Discussion by rgarcia89** [thanos-io/thanos#6637] [Questions & Answers]
Hi, I am currently alerting when the GRPC Client Error Rate for one of the Thanos Query Endpoints is rising. However, in this alert I am not seeing which endpoint exactly is causing the issue. Does anyone know if there is a metric which I could use, in order to obtain these information?

Current alert definition
```
        - alert: ThanosQueryGrpcClientErrorRate
          annotations:
            description: Thanos Query {{$labels.job}} is failing to send {{$value | humanize}}% of requests.
            runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanosquerygrpcclienterrorrate
            summary: Thanos Query is failing to send requests.
          expr: |
            (
              sum by (job) (rate(grpc_client_handled_total{grpc_code!="OK", job=~".*thanos-query.*"}[5m]))
            /
              sum by (job) (rate(grpc_client_started_total{job=~".*thanos-query.*"}[5m]))
            ) * 100 > 5
```

**Comment by saswatamcode** [comment]
The `grpc_method` and `grpc_service` methods should give you a clue! Try to group by those labels.

**Reply by rgarcia89** [reply]
@saswatamcode what I am looking for to be honest are metrics that make me link the issue to the specific sidecar endpoint which is causing it

![Screenshot from 2023-08-29 11-07-57](https://github.com/thanos-io/thanos/assets/4134730/4d00af01-d753-4105-a6e2-10b1f5c33b9a)

**Reply by mikejoh** [reply]
@rgarcia89 Sorry for necroposting, but i'm looking for something similar, a metric that could show me if the sidecar endpoint is up or down. Did you find anything regarding this? Thanks!

**Reply by rgarcia89** [reply]
Hi @mikejoh, unfortunately not
